### PR TITLE
[Navigation] update KotlinPoet to 1.8.0 for navigation safe-args

### PR DESCRIPTION
**Pull request by tatocaster**
## Proposed Changes

Update KotlinPoet to 1.8.0 for navigation safe-args.

## Testing

Test: ./gradlew test

## Issues Fixed

Fixes: The bug on [https://issuetracker.google.com/issues/183990444](https://issuetracker.google.com/issues/183990444) being fixed

### CODE REVIEW

**Comment by tatocaster** [comment]
except the PR, 

``` kotlin
const val KOTLINPOET = "com.squareup:kotlinpoet:1.4.0"
const val KOTLINPOET_METADATA = "com.squareup:kotlinpoet-metadata:1.4.0"
const val KOTLINPOET_METADATA_SPECS = "com.squareup:kotlinpoet-metadata-specs:1.4.0"
const val KOTLINPOET_CLASSINSPECTOR_ELEMENTS = "com.squareup:kotlinpoet-classinspector-elements:1.4.0"
``` 
Those properties are not used anywhere. I guess we can safely remove

File: navigation/navigation-safe-args-generator/build.gradle [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies {
     implementation(KOTLIN_STDLIB)
 
     implementation(JAVAPOET)
-    implementation("com.squareup:kotlinpoet:1.7.2")
+    implementation("com.squareup:kotlinpoet:1.8.0")
```
Comment (ianhanniballake) [inline navigation/navigation-safe-args-generator/build.gradle]: Use the `KOTLINPOET` variable here and update the value there.

**Review by ianhanniballake — CHANGES_REQUESTED** [review]
As part of this, can you please remove the unused dependencies:

> const val KOTLINPOET_METADATA = "com.squareup:kotlinpoet-metadata:1.4.0"
> const val KOTLINPOET_METADATA_SPECS = "com.squareup:kotlinpoet-metadata-specs:1.4.0"
> const val KOTLINPOET_CLASSINSPECTOR_ELEMENTS = "com.squareup:kotlinpoet-classinspector-elements:1.4.0"

**Review by ianhanniballake — APPROVED** [review]

**Comment by ianhanniballake** [comment]
Thanks for the pull request!

**Pull request closed without merging**